src/oubliette2john.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+
+import sys
+import struct
+import binascii
+
+def process_file(filename):
+    try:
+        with open(filename, 'rb') as f:
+            # Read and validate header
+            header = f.read(12)
+            if len(header) != 12:
+                sys.stderr.write(f"{filename}: Invalid file size\n")
+                return -1
+
+            # Unpack header fields
+            magic = header[0:5].decode('ascii')
+            major_ver = header[5]
+            minor_ver = header[6]
+            algorithm = header[7]
+            data_size = struct.unpack("<I", header[8:12])[0]
+
+            if magic != "OUBPF":
+                sys.stderr.write(f"{filename}: Invalid magic bytes\n")
+                return -2
+
+            if algorithm not in [0, 1]:
+                sys.stderr.write(f"{filename}: Unknown algorithm {algorithm}\n")
+                return -3
+
+            # Read the PasswordEncryptedHash
+            password_hash = f.read(32)
+            if len(password_hash) != 32:
+                sys.stderr.write(f"{filename}: Failed to read PasswordEncryptedHash\n")
+                return -4
+
+            # Format output for John the Ripper with different tags for Blowfish and IDEA
+            hex_hash = binascii.hexlify(password_hash).decode('ascii')
+            format_tag = "oubliette-blowfish" if algorithm == 0 else "oubliette-idea"
+            print(f"${format_tag}${major_ver}.{minor_ver}${hex_hash}")
+            return 0
+
+    except IOError as e:
+        sys.stderr.write(f"{filename}: {str(e)}\n")
```

Comment:
`str()` function is unnecessary here and will be called implicitly anyway. `{e}` is enough for simplification.